icubam/www/handlers/update.py

```diff
@@ -28,18 +28,15 @@ def get_consent_html(self, user):
         return fp.read()
 
   def get_current_user(self):
-    # There is a cookie. we use it.
-    if self.get_secure_cookie(self.COOKIE):
-      return super().get_current_user()
-    else:  # the user is coming on his own via the link with token.
-      return "unknown_user"
+    """This route is not secured at first."""
+    return None
 
   async def get(self):
     """Serves the page with a form to be filled by the user."""
     user_token = self.get_query_argument(self.QUERY_ARG)
     user, icu = self.decode_token(user_token)
     if user is None or icu is None:
-      logging.error(f"Token authentication failed")
+      logging.error("Token authentication failed")
       self.clear_cookie(self.COOKIE)
       return self.set_status(404)
```

Comment:
This should be a 403, not 404.

Reply:
actually we want to keep it a 404, as it the route does not exists... but we can change later